pkg/ansible/proxy/cache_response.go

```diff
@@ -227,13 +238,24 @@ func (c *cacheResponseHandler) getObjectFromCache(r *requestfactory.RequestInfo,
 	un := &unstructured.Unstructured{}
 	un.SetGroupVersionKind(k)
 	obj := client.ObjectKey{Namespace: r.Namespace, Name: r.Name}
-	err := c.informerCache.Get(context.Background(), obj, un)
-	if err != nil {
-		// break here in case resource doesn't exist in cache but exists on APIserver
-		// This is very unlikely but provides user with expected 404
-		log.Info(fmt.Sprintf("Cache miss: %v, %v", k, obj))
-		return nil, err
+	errChan := make(chan error, 1)
+	go func() {
+		err := c.informerCache.Get(context.Background(), obj, un)
+		errChan <- err
+	}()
+
+	select {
+	case watchErr := <-errChan:
+		if watchErr != nil {
+			// break here in case resource doesn't exist in cache but exists on APIserver
+			// This is very unlikely but provides user with expected 404
+			log.Info(fmt.Sprintf("cache miss: %v err-%v", k, watchErr))
+			return nil, watchErr
+		}
+	case <-time.After(cacheEstacacheEstablishmentTimeout):
+		return nil, fmt.Errorf("timeout establishing watch, commonly permissions of the controller are not sufficent")
 	}
+
```

Comment:
Rather than using a channel, a goroutine, and a timer, a more idiomatic approach would be to use `context.WithTimeout(context.Background(), cacheEstablishmentTimeout)`

And then check the error from the `Get()` call to see if it is a `context.DeadlineExceeded` error.

Ditto for the `informerCache.List()` call above.